### ISSUE TITLE
[Feature] T.tile.fill不支持动态切片

### ISSUE THREAD

**Issue by fengz72** [tile-ai/tilelang-ascend#1207]
## Feature Description
需要根据运行时计算出来的长度, 动态的指定fill填充的长度
```python
idx = T.if_then_else(A[0, 0] > 0, 32, 64)
T.tile.fill(c_ub[0, 0: idx], 1.0)
```

## Use Case
```
import argparse

import tilelang
import tilelang.language as T
import torch

tilelang.cache.clear_cache()

parser = argparse.ArgumentParser(description="NPU Kernel Compilation")
parser.add_argument("--m", type=int, default=1024, help="Matrix M dimension")
parser.add_argument("--n", type=int, default=1024, help="Matrix N dimension")
args = parser.parse_args()

M = args.m
N = args.n


@tilelang.jit(out_idx=[-1])
def vec_add(M, N, block_M, block_N, dtype="float"):
    m_num = M // block_M
    n_num = N // block_N

    VEC_NUM = 2

    @T.prim_func
    def main(
            A: T.Tensor((M, N), dtype),
            B: T.Tensor((M, N), dtype),
            C: T.Tensor((M, N), dtype),
    ):
        with T.Kernel(m_num * n_num, is_npu=True) as (cid, vid):
            bx = cid // n_num
            by = cid % n_num

            a_ub = T.alloc_ub((block_M // VEC_NUM, block_N), dtype)
            b_ub = T.alloc_ub((block_M // VEC_NUM, block_N), dtype)
            c_ub = T.alloc_ub((block_M // VEC_NUM, block_N), dtype)
            with T.Scope("V"):
                T.copy(A[bx * block_M + vid * block_M // VEC_NUM, by * block_N], a_ub)
                T.copy(B[bx * block_M + vid * block_M // VEC_NUM, by * block_N], b_ub)

                T.barrier_all()
                idx = T.if_then_else(A[0, 0] > 0, 32, 64)
                T.tile.fill(c_ub[0, 0: idx], 1.0)
                T.barrier_all()

                T.copy(c_ub, C[bx * block_M + vid * block_M // VEC_NUM, by * block_N])

    return main


func = vec_add(M, N, 128, 256)

torch.manual_seed(0)

a = torch.randn(M, N).npu()
b = torch.randn(M, N).npu()

torch.npu.synchronize()
print("init successful!")

c = func(a, b)
```

## Proposed Solution
有两个地方需要更改
1. `tilelang/language/ascend_tile.py`
```python
from functools import reduce
import operator
from tvm.tir.expr import PrimExpr

def _handle_buffer_region(br: BufferRegion, mask):
    bf = br.buffer
    indices = [x.min for x in br.region]
    offset = bf.offset_of(indices)[0]
    extent = [x.extent for x in br.region]
    if any(isinstance(e, PrimExpr) for e in extent):
        size_extent = reduce(operator.mul, extent)
    else:
        size_extent = math.prod(extent)
    return bf.access_ptr(mask, offset=offset, extent=size_extent), extent
```

```python
def fill(buffer: Buffer | BufferRegion, value: PrimExpr):
    """Fill a buffer or buffer region with a specified value.

    Args:
        buffer: Either a TVM buffer or buffer region to be filled
        value: The value to fill the buffer with

    Returns:
        A TVM intrinsic call that performs the fill operation
    """
    if isinstance(buffer, BufferRegion):
        buffer_ptr, buffer_extent = _handle_buffer_region(buffer, "w")
        if any(isinstance(e, PrimExpr) for e in buffer_extent):
            size = reduce(operator.mul, buffer_extent)
        else:
            size = math.prod(buffer_extent)
    else:
        buffer_ptr = buffer.access_ptr("w")
        size = math.prod(buffer.shape)

    return tir.call_intrin(
        "handle",
        tir.op.Op.get("tl.ascend_fill"),
        f"Fill<{_dtype(buffer)}>",
        buffer_ptr,
        value,
        size,
    )
```

2. `3rdparty/tvm/python/tvm/tir/buffer.py`
```bash
diff --git a/python/tvm/tir/buffer.py b/python/tvm/tir/buffer.py
index ec57ad780..262d6adcb 100644
--- a/python/tvm/tir/buffer.py
+++ b/python/tvm/tir/buffer.py
@@ -185,7 +185,7 @@ class Buffer(Object, Scriptable):
         if not isinstance(indices, (tuple, list)):
             indices = [indices]
         has_slice = any(isinstance(i, slice) for i in indices)
-        has_step = any(isinstance(i, slice) and i.step is not None for i in indices)
+        has_step = any(isinstance(i, slice) and i.step is not None and i.step != 1 for i in indices)
         analyzer = Analyzer()
         if has_slice and not has_step:
             region = []
```

## Feature Type
- [x] New operator/kernel (e.g., new GEMM variant, new attention mechanism)
- [ ] New API/primitive (e.g., new T.xxx operation)
- [ ] Compiler optimization (e.g., new pass, performance improvement)
- [ ] New backend support (e.g., new hardware target)
- [ ] Developer tooling (e.g., debugging, profiling)
- [ ] Documentation improvement
- [ ] Other: [specify]

## Operator Details (if applicable)

**Operator Type:**
[e.g. GEMM, Flash Attention, Softmax, LayerNorm, Convolution, etc.]

**Input/Output Specification:**
```
Input shapes and dtypes:
- A: [M, K], float16
- B: [K, N], float16

Output shapes and dtypes:
- C: [M, N], float16
```

**Performance Requirements:**
[e.g. target throughput, latency, memory bandwidth]

**Reference Implementation (if any):**
[e.g. PyTorch implementation, CUDA kernel, paper reference]

## Alternative Solutions
Describe any alternative solutions or features you've considered.

## Implementation Constraints (if known)

**Programming Mode Preference:**
- [ ] Developer mode (automatic)
- [ ] Expert mode (manual control)
- [x] Both modes supported

**Memory Constraints:**
[e.g. L1 buffer size, UB buffer size constraints]

**Target Hardware:**
- [x] Ascend A2
- [x] Ascend A3
- [ ] Other Ascend variants

## Additional Context
Add any other context, references, or screenshots about the feature request here.

## Willingness to Contribute
- [ ] I'm willing to submit a PR for this feature
- [x] I can help with testing
- [x] I can provide reference implementations
- [ ] I need guidance to get started

**Comment by fengz72** [comment]
改写为`T.tile.fill(c_ub[0, : idx], 1.0)`后编译可以通过

**Comment by fengz72** [comment]
https://github.com/apache/tvm/blob/main/python/tvm/tirx/buffer.py, tvm已经更新, 解决此bug

**Comment by liggest** [comment]
> https://github.com/apache/tvm/blob/main/python/tvm/tirx/buffer.py, tvm已经更新, 解决此bug

[[TIRx] Bringup TIRx Infrastructure (apache/tvm@859498d)](https://github.com/apache/tvm/commit/859498dc01660d06f61d54c98edf27828d68f25e#diff-89ddf286895419979dea0e747b22ca611aed7f350650722eb9c4571c88ffd818)
具体 commit 点好像是这个

**Comment by liggest** [comment]
> 改写为`T.tile.fill(c_ub[0, : idx], 1.0)`后编译可以通过

编译通过后可能还有精度问题，再观察一下 🤔

**Comment by fengz72** [comment]
tilelang/language/ascend_tile.py经过测试不需要修改

**Comment by fengz72** [comment]
已验证修复
